packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md

```diff
@@ -1,36 +1,42 @@
-Check for an existing session in two steps.
+To check for an existing session may take two steps:
 
-1. Initialize the authorization state manager (`authStateManager`) from a saved session.
-2. Verify that the access token is still valid by checking the expiry.
+- Check for an existing valid token.
+- Refresh the token if it has expired.
 
-The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
 
-1. Check for an expired token by updating the `tokenExpired` utility function:
+Add an `.onAppear` modifier to the main content view that checks for an existing session:
 
-   ```swift
-   func tokenExpired(_ tokenString: String?) -> Bool {
-     guard let accessToken = tokenString,
-     let tokenInfo = OKTIDToken.init(idTokenString: accessToken) else {
-       return false
-     }
-      
-     return Date() > tokenInfo.expiresAt
+```swift
+var body: some View {
+   ZStack {
+      ...
+            ProgressView("Waiting")
+         }
+      }
+      .onAppear() {
+         busy = true
+         // Check for an existing valid token.
+         if let credential = Credential.default {
+            Task {
+               do {
+                  // Make sure the token is valid.
+                  try await credential.refreshIfNeeded()
+                  // The token is valid, update the app state.
+                  isSignedIn = true
+               }
+               catch {
+                  // The user wasn't signed in or the token couldn't be refreshed.
+                  // No action is required.
+               }
+               // The token is invalid, hide the activity indicator.
+               busy = false
+            }
+         } else {
+            // There is existing session, hide the activity indicator.
```

Comment:
```suggestion
            // If there is an existing session, hide the activity indicator.
```

Reply:
A great catch, though it may be different text is required. The semantics of this else clause are that there is no existing session. Perhaps a comment of:
// There is no existing session, hide the activity indicator

Reply:
makes perfect sense to me, thanks.